extensions/gdx-box2d/gdx-box2d/src/com/badlogic/gdx/physics/box2d/World.java

```
@@ -91,17 +92,21 @@ class CustomContactFilter: public b2ContactFilter
 private:
 	JNIEnv* env;
 	jobject obj;
+	jboolean useDefaultContactFilter;
 
 public:
 	CustomContactFilter( JNIEnv* env, jobject obj )
 	{
 		this->env = env;
 		this->obj = obj;
+		useDefaultContactFilter = env->CallBooleanMethod( obj, useDefaultContactFilterID );
```

This needs a check whether `useDefaultContactFilterID != 0`.
If proguard/r8/native-image strips the method, this might lead to a segfault otherwise.

---

> This needs a check whether `useDefaultContactFilterID != 0`. If proguard/r8/native-image strips the method, this might lead to a segfault otherwise.

I've tested and it currently throws a Java `NoSuchMethodException` if Proguard doesn't add it. Won't adding that check swallow the error and the user won't notice as Java version would be executed in that case?

---

When is the error thrown? When getting the methodID or when trying to call the actual method?
Because I know native-image just segfaults if methods aren't registered for JNI.

---

I'd guess it's thrown when trying to execute the actual method as it's classic normal `NoSuchMethodException` Java Exception that specifies the method name in the stacktrace.

---

```
 FATAL EXCEPTION: GLThread 138
                                                                                                    Process: com.xxxx.xxxx, PID: 16140
                                                                                                    java.lang.NoSuchMethodError: no non-static method "Lcom/badlogic/gdx/physics/box2d/World;.getUseDefaultContactFilter()Z"
                                                                                                    	at com.badlogic.gdx.physics.box2d.World.newWorld(Native Method)
                                                                                                    	at com.badlogic.gdx.physics.box2d.World.<init>(SourceFile:122)
                                                                                                    	at t6.g.f(SourceFile:11)
                                                                                                    	at c7.a.l(SourceFile:1)
                                                                                                    	at k7.b0.m(SourceFile:305)
                                                                                                    	at t6.q.a(SourceFile:135)
                                                                                                    	at k7.a.l(SourceFile:3)
                                                                                                    	at k7.r0.a(SourceFile:35)
                                                                                                    	at j7.j.l(SourceFile:11)
                                                                                                    	at k2.e.k(SourceFile:46)
                                                                                                    	at j7.j.k(SourceFile:1)
                                                                                                    	at h2.g.a(SourceFile:103)
                                                                                                    	at h2.h.h(SourceFile:97)
                                                                                                    	at g1.g.h(SourceFile:17)
                                                                                                    	at l1.s.f(SourceFile:196)
                                                                                                    	at l1.i.onDrawFrame(SourceFile:180)
                                                                                                    	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1573)
                                                                                                    	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1272)
```

---

It's ehen calling the Native one :D

---

>  It's when getting the method id :D

Ahh, I see! Then I think we could add a "0" check should be added, just to be safe in case the world is still getting used for some reason, but I have no strong opinion.

---

> Ahh, I see! Then I think we could add a "0" check should be added, just to be safe in case the world is still getting used for some reason, but I have no strong opinion.

Ok, I've added the check even if it should never reach that point with it being 0 as the informative Error will have been thrown before.